### ISSUE TITLE
Augmented input at time T based on output at time T-1

### ISSUE THREAD

**Issue by suryabhupa** [jcjohnson/torch-rnn#155]
Normally, when decoding with an LSTM, the output at time t-1 is used as the input at time t. Instead, I want the input at time t to be [output_{t-1} f(output_{t-1})], where f is feedforward neural network. Does anyone know how to cleanly incorporate this into the code?

**Comment by jcjohnson** [comment]
I'm guessing that you want to do something like spatial attention in [Show, Attend, and Tell](https://arxiv.org/abs/1502.03044)? In that case f takes the hidden state at time t-1, computes an attention mask over the input, computes a weighted average of spatial features using that attention mask, and feeds the result back to the RNN at time t.

This will be nontrivial to add to the current code; you'll basically need to rewrite the `nn.LSTM` layer from scratch.

**Comment by jcjohnson** [comment]
For more complex recurrence patterns like this, you may have better luck with [torch-rnnlib](https://github.com/facebookresearch/torch-rnnlib) or [Element-Research/rnn](https://github.com/Element-Research/rnn).

**Comment by suryabhupa** [comment]
Actually, I was looking for something even simpler than computing spatial attention over the inputs (although this is probably most people's use case). f would be a function that encodes the 'difference' between o_t-1 or [o_t-1 ... o_1] and the complete target output sequence O, and the next decoding step is then conditioned on this as well as o_t-1. Do you think this is still too complex to be neatly integrated into the code?

**Comment by suryabhupa** [comment]
In fact, I have an even simpler (and possibly dumb) question -- how do I do decoding without teacher forcing?

At test time, I would like to use the output at t-1 as the input at t, but I want to simply provide the initial hidden and cell state to the LSTM and decode until a EOS token is hit. What is the right way to do this?
